File: src/main/java/io/openfuture/openmessanger/service/impl/PrivateChatServiceImpl.java
package io.openfuture.openmessanger.service.impl;

import java.util.List;
import java.util.Optional;

import org.springframework.stereotype.Service;

import io.openfuture.openmessanger.repository.ChatParticipantRepository;
import io.openfuture.openmessanger.repository.entity.ChatParticipant;
import io.openfuture.openmessanger.service.PrivateChatService;
import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
public class PrivateChatServiceImpl implements PrivateChatService {

    private final ChatParticipantRepository chatParticipantRepository;

    @Override
    public ChatParticipant getOtherUser(final String username, final Integer chatId) {
        final List<ChatParticipant> participants = chatParticipantRepository.findAllByChatId(chatId);
        if (participants.size() == 1) {
            return participants.get(0);
        }
        final Optional<ChatParticipant> recipient = participants.stream()
                                                                .filter(chatParticipant -> !chatParticipant.getUsername().equals(username))
                                                                .findFirst();
        return recipient.orElseThrow(() -> new IllegalStateException("No other participants of chat"));
    }

}

File: src/main/kotlin/io/openfuture/openmessanger/service/impl/PrivateChatServiceImpl.kt
package io.openfuture.openmessanger.service.impl

import io.openfuture.openmessanger.repository.ChatParticipantRepository
import io.openfuture.openmessanger.repository.entity.ChatParticipant
import io.openfuture.openmessanger.service.PrivateChatService
import lombok.RequiredArgsConstructor
import org.springframework.stereotype.Service

@Service
@RequiredArgsConstructor
class PrivateChatServiceImpl(
    val chatParticipantRepository: ChatParticipantRepository
) : PrivateChatService {

    override fun getOtherUser(username: String, chatId: Int?): ChatParticipant? {
        val participants = chatParticipantRepository.findAllByChatId(chatId)
        if (participants!!.size == 1) {
            return participants[0]
        }
        val recipient = participants.stream()
            .filter { chatParticipant: ChatParticipant? -> chatParticipant?.username != username }
            .findFirst()
        return recipient.orElseThrow { IllegalStateException("No other participants of chat") }
    }
}